bpf/lib/nat.h

```diff
@@ -515,9 +534,19 @@ static __always_inline __maybe_unused int snat_v4_create_dsr(struct __ctx_buff *
 	return CTX_ACT_OK;
 }
 
+static __always_inline void snat_v4_init_tuple(const struct iphdr *ip4,
+					       enum nat_dir dir,
+					       struct ipv4_ct_tuple *tuple)
+{
+	tuple->nexthdr = ip4->protocol;
+	tuple->daddr = ip4->daddr;
+	tuple->saddr = ip4->saddr;
+	tuple->flags = dir;
```

Comment:
Thanks for the change. However, I mean to move the whole tuple extraction into the helper - not only L3, but also L4 (port extraction / ICMP ID is still outside the helper).

Reply:
Hum... I would have appreciated more details in the previous set of comments ;-)  As mentioned I was mitigated doing this change and now understanding that you was also requesting L4 changes but I don't think that will work.

We are not expecting nat and rev-nat to share the same kind of features. For instance, the support of ICMP DestUnreach need to frag only makes sense for rev-nat.